core/src/main/java/org/apache/carbondata/core/util/path/CarbonTablePath.java

```diff
@@ -665,9 +665,18 @@ public static String getCarbonIndexFileName(String actualBlockName) {
    * @return
    */
   public static String getShardName(String actualBlockName) {
-    return DataFileUtil.getTaskNo(actualBlockName) + "-" + DataFileUtil.getBucketNo(actualBlockName)
-        + "-" + DataFileUtil.getSegmentNo(actualBlockName) + "-" + DataFileUtil
-        .getTimeStampFromFileName(actualBlockName);
+    String segmentNoStr = DataFileUtil.getSegmentNo(actualBlockName);
+    // data before version 1.4 does not have segmentId
+    if (null != segmentNoStr) {
+      return DataFileUtil.getTaskNo(actualBlockName) + "-" +
```

Comment:
better to use stringbuilder if concat more than 3 strings

Reply:
fixed